### PR TITLE
gnomeExtensions.wsmatrix:  new package

### PR DESCRIPTION
**Pull request by deepfire**
Add the wsmatrix extension to Ghome Shell (compatible with v3.32):

- https://extensions.gnome.org/extension/1485/workspace-matrix/

###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nix-review --run "nix-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

###### Notify maintainers

cc @

### CODE REVIEW

**Comment by deepfire** [comment]
cc-ing some gnome shell extension maintainers: @aneeshusa @jonafato @svsdep @tiramiseb @timbertson

**Comment by deepfire** [comment]
I think we might also deprecate `workspace-grid`, since it's:

1. not even available (its nix definition isn't imported)
2. unmaintained, with `wsmatrix` taking it over.

File: pkgs/desktops/gnome-3/extensions/wsmatrix.nix
Comment (jonafato) [inline pkgs/desktops/gnome-3/extensions/wsmatrix.nix]: Can you specify a release tag from https://github.com/mzur/gnome-shell-wsmatrix/releases as the version?

File: pkgs/desktops/gnome-3/extensions/wsmatrix.nix
Comment (jonafato) [inline pkgs/desktops/gnome-3/extensions/wsmatrix.nix]: Most of the gnome shell extensions use the convention `gnome-shell-extension-...` instead of `gnome-shell-...`. I don't have an opinion here other than "consistency is good", so unless others object, please add `-extension` here.

File: pkgs/top-level/all-packages.nix
Comment (jonafato) [inline pkgs/top-level/all-packages.nix]: Can you move `wsmatrix.nix` to `wsmatrix/default.nix` to match the convention of the other extensions?

**Review by jonafato — CHANGES_REQUESTED** [review]
The extension seems to work well, just a few requests to match naming and version conventions.

**Comment by jonafato** [comment]
>I think we might also deprecate workspace-grid, since it's:
>
>1.    not even available (its nix definition isn't imported)
>2.    unmaintained, with wsmatrix taking it over.

Reference for the unmaintained statement: https://github.com/workspace-grid/workspace-grid/issues/120. I can't speak to the maintained status of it, though there's some discussion on that thread. I'd typically be against removing it without a definite unmaintained status, but given your reasoning plus the fact that it doesn't work in 3.32 (the only gnome version currently supported in nixpkgs), there doesn't seem to be much harm. That said, I'll defer to @aneeshusa on removing it since they're listed as the maintainer.

**Comment by chkno** [comment]
Oops: I didn't see this open PR and submitted a ~duplicate one: #75123 .  That one addresses the above review comments, if it's easier to continue over there.

**Comment by jonafato** [comment]
Looks like this was added in #75123, so this PR should be closed.

**Pull request closed without merging**
